File: src/01-easy/temperatures/solution.go
package main
import (
    "fmt"
    "math"
)
func main() {
    var n int
    fmt.Scan(&n)
    if n == 0 { fmt.Println(0); return }
    r := 0.0
    for i := 0; i < n; i++ {
        var v int
        fmt.Scan(&v)
        vf := float64(v)
        if i == 0 || math.Abs(vf) < math.Abs(r) || (math.Abs(vf) == math.Abs(r) && vf > 0) { r = vf }
    }
    fmt.Println(int(r))
}

File: src/01-easy/temperatures/solution.java
import java.util.Scanner;
class Solution {
    public static void main(String[] a) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        if (n == 0) { System.out.println(0); return; }
        int r = s.nextInt();
        for (int i = 1; i < n; i++) {
            int v = s.nextInt();
            if (Math.abs(v) < Math.abs(r) || (Math.abs(v) == Math.abs(r) && v > 0)) r = v;
        }
        System.out.println(r);
    }
}
